src/phases/abstract-form-change-base-phase.ts

```diff
@@ -50,7 +50,7 @@ export abstract class FormChangeBasePhase extends Phase {
 
       globalScene.audioManager.fadeOutBgm(undefined, false);
 
-      this.handler = ui.getHandler() as FormChangeSceneUiHandler;
+      this.handler = ui.getCurrentHandler() as FormChangeSceneUiHandler;
```

Comment:
NIT: Use generic param `<T>`
```suggestion
      this.handler = ui.getCurrentHandler<FormChangeSceneUiHandler>();
```